### PR TITLE
Memory Tag Editing Functionality via Dashboard Map

### PR DESCRIPTION
**Pull request by aqibisbiqa**
# Description

This PR allows users to update agent memory values using the dashboard map UI. 

Frontend changes:
- The map ui is denser and stylings modified to match rest of dashboard
- Table for each object has a set of immutable fields (i.e. memid, unique keys) that cannot be changed. All other fields are mutable, denoted with a black input box
- Input
- Red clear icon in top right closes table and doesn't send any changes
- Submit icon only enabled when **valid** changes made. Clicking closes table and sends changes
- Refresh icon resets all values to what they were when object was clicked
- All edits handled via the editManager (a dictionary of edits), filtered editManager sent to agent where processed in perception module

Backend:
- New perception module `ManualEditsPerception` introduced to handle all user made edits to agent memory
- This module perceives after all other perception modules, so its effects overwrites all others

Quirks:
- If object has both "location" and "pos" attributes, only changing "pos" will result in change
- Changing a spatial attribute like "pos" will result in object appearing static on map. Not issue for objects meant to be static, but if agent "pos" is altered it will appear static on map while moving like normal in minecraft/voxel world

Fixes # (issue)

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [ ] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [x] I want a deep design review.

## Before and After

Before:
Screen Shot 2022-07-06 at 3 50 00 PM

After:
Screen Shot 2022-07-06 at 3 51 08 PM
Error Handling
Screen Shot 2022-07-06 at 4 03 05 PM
Can hover to get full value if shortened
Screen Shot 2022-07-06 at 4 04 42 PM
Refresh restores values
Screen Shot 2022-07-06 at 4 06 05 PM

# Testing

You can test these changes on my branch `aqib/memory_tag_edits`. To run I used `python droidlet/lowlevel/minecraft/cuberite_process.py --config really_flat_world` and `python agents/craftassist/craftassist_agent.py --draw_map --enable_timeline`. The only flag you really need though is `--draw_map` with the agent, let me know if things don't work with certain flags/options.

The table might be slightly over-engineered and has several features built in. I've tried to address a couple edge cases without making the code too cluttered, although others that will use this table should try making edits and seeing if they run into issues. Prolonged usage is the only way to really test this feature.

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [x] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [x] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I ran on hardware (1) all scripts in `tests/scripts`, (2) asv benchmarks.

### CODE REVIEW

File: droidlet/dashboard/web/src/components/Memory2D.js [hunk 1]
```diff
@@ -275,6 +275,7 @@ class Memory2D extends React.Component {
             fill={color}
             onClick={(e) => {
               this.handleObjClick("obstacle_map", x, y, {
+                memid: "don't edit",
```
Comment (aqibisbiqa) [inline droidlet/dashboard/web/src/components/Memory2D.js:278]: These objects refer to those obtained from `get_obstacle_list` defined in `droidlet/memory/place_field`. I believe some examples of these objects are "hole" or "pit" that are initialized with the world. I don't know how we should deal with these on the map (they aren't drawn from ReferenceObjects), so I left this "don't edit" placeholder

File: droidlet/dashboard/web/src/components/Memory2D/MemoryMapTable.js [hunk 2]
```diff
@@ -52,71 +119,139 @@ const useStyles = makeStyles({
...
  */
 export default function MemoryMapTable(props) {
-  const classes = useStyles();
+  /*
+  The editManager handles the state of the table in the form of a dictionary of the rows.
+  Keys are row attributes.
+  Value is object [
+    value: <value>,
+    valueType: <value_type>,
+    status: "same" or "changed" or "error" [or "new" or "deleted"] [for future],
```
Comment (aqibisbiqa) [inline droidlet/dashboard/web/src/components/Memory2D/MemoryMapTable.js:128]: If we want to include functionality for adding/removing tags, we can alter "status" and handle with ManualEditsPerception module in backend

**Review by ys1382fb — APPROVED** [review]
